osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -146,34 +135,56 @@ private void applyRandomisation(float rateOfChangeMultiplier, RandomObjectInfo p
         /// </summary>
         private void moveSliderIntoPlayfield(Slider slider, RandomObjectInfo currentObjectInfo)
         {
-            // Min. distances from the slider's position to the playfield border
+            var minMargin = getMinSliderMargin(slider);
+
+            slider.Position = new Vector2(
+                Math.Clamp(slider.Position.X, minMargin.Left, OsuPlayfield.BASE_SIZE.X - minMargin.Right),
+                Math.Clamp(slider.Position.Y, minMargin.Top, OsuPlayfield.BASE_SIZE.Y - minMargin.Bottom)
+            );
+
+            currentObjectInfo.PositionRandomised = slider.Position;
+            currentObjectInfo.EndPositionRandomised = slider.EndPosition;
+
+            shiftNestedObjects(slider, currentObjectInfo.PositionRandomised - currentObjectInfo.PositionOriginal);
+        }
+
+        /// <summary>
+        /// Calculates the min. distances from the <see cref="Slider"/>'s position to the playfield border for the slider to be fully inside of the playfield.
+        /// </summary>
+        private MarginPadding getMinSliderMargin(Slider slider)
+        {
             var minMargin = new MarginPadding();
+            Vector2 pos;
 
-            foreach (var hitObject in slider.NestedHitObjects.Where(o => o is SliderTick || o is SliderEndCircle))
+            for (double i = 0; i <= 1; i += 1 / (slider_path_checking_rate / 1000 * (slider.EndTime - slider.StartTime)))
             {
-                if (!(hitObject is OsuHitObject osuHitObject))
-                    continue;
+                pos = slider.Path.PositionAt(i);
+                updateMargin();
+            }
```

Comment:
1. This is essentially computing axis-aligned bounding box right? I think computing the bounding box first then deriving the minimum margin from that is easier to follow.
2. Instead of getting a vertex fixed time increment, get vertices of the line segments by `SliderPath.GetPathToProgress`. More efficient and precise.